### PR TITLE
Add OAuth2 docs

### PR DESCRIPTION
**Pull request by rullzer**
A first start.

@MariusBluem could you take over an finish this (make the screenshots?) maybe make it look a bit better?

### CODE REVIEW

**Comment by MariusBluem** [comment]
On my ToDo.

**Comment by MariusBluem** [comment]
@rullzer Have added a screenshot and did some smaller text improvements... Please have a look, then we can merge this one :) Thanks for your patience 😅

**Comment by MariusBluem** [comment]
This will fix #749 .. Please review @MorrisJobke @rullzer

**Comment by MorrisJobke** [comment]
Rebased to re-run CI.

File: admin_manual/configuration_server/oauth2.rst [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+OAuth2 specification.
+
+.. note:: Nextcloud does only support confidential clients.
+
+Add an OAuth2 Application
+-------------------------
+
+Head over to your Administrator Security Settings. Here you can add a new ``OAuth2`` client.
+
+.. figure:: ../images/oauth2-settings.png
```
Comment (MorrisJobke) [inline admin_manual/configuration_server/oauth2.rst]: I guess the `../` is not needed here:

```
LaTeX Warning: File `{configuration_server/../images/oauth2-settings}.png' not found on input line 7788.
```

**Comment by MorrisJobke** [comment]
I guess we should port this to 13, right?

**Comment by MorrisJobke** [comment]
stable13 fc78d9e4 12396102 e481a023

**Pull request merged**